source/common/json/json_loader.cc

```diff
@@ -521,7 +523,7 @@ bool ObjectHandler::StartObject() {
     state_ = expectKeyOrEndObject;
     return true;
   default:
```

Comment:
This case must crash the process, so throwing a Json::Exception, or any exception, is not appropriate. Especially if it cannot be tested without modifying the rapidjson code. Using NOT_REACHED is fine in this case.